backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,494 @@
+                        "batch_size": 45,
+                        "device_count": 0,
+                    }
+                ],
+            }),
+        # Phase, Five batches, just make sure it works. Should behave like all other > 1 cases
+        (
+            {
+                "name":
+                "Five phases, first no start time, last start time, no batch size",
+                "artifact_name":
+                "deployments-phase-testing",
+                "devices": ["dummyuuid" + str(i) for i in range(100)],
+                "phases": [
+                    {
+                        # Start time is optional in the first stage (default to now)
+                        "batch_size": 10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=2)).isoformat("T")) + "Z",
+                        # Batch size is optional in the last stage (ie, it is the remaining devices)
+                    },
+                ],
+            },
+            {
+                "name":
+                "Five phases, first no start time, last start time, no batch size",
+                "artifact_name":
+                "deployments-phase-testing",
+                "device_count":
+                100,
+                "phases": [{
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 60,
+                    "device_count": 0,
+                }],
+            }),
+    ]
+
+    @pytest.mark.parametrize("test_case", test_cases)
+    def test_phased_deployments_success(self, test_case,
+                                        setup_deployments_enterprise_test):
+
+        user, tenant, utoken = setup_deployments_enterprise_test
```

Comment:
I can see you create another tenant in setup_..., so that's cool.

but it's not used ever, right? it should be.

the logic should look like this:
* always have two tenants (one user per each)
* parametrize the whole test with a tenant - so that it's ran twice
* in the test - do the test for *this* tenant, but also - check that the other tenant's deployment count is still 3

as a result we have tested that evt works in multitenancy, regardless of tenant name.

you'll have to return tenants + users + tokens from setup. they can be just dicts `{tenant_name, user_name, user_token} x2` and you're all set. sometimes we use structs Tenant and User from common too but that's optional.

you already know how to use `pytest.mark.parametrize` so you're set.